### PR TITLE
[RFR] Unit testing

### PR DESCRIPTION
**Pull request by Kmaschta**
- [x] ~~Unit testing fundations~~
- [x] ~~composer.json update~~
- [x] ~~autoload config fix~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,3 +16,9 @@ Pending the final .phar file, you can use symfony command :
 ```bash
 php src/application.php dobble:run -h
 ```
+
+# Unit tests
+
+```bash
+make install
```
Comment (fzaninotto) [inline README.md]: no need to repeat the `make install` here, it's implied.

File: tests/DobbleCommandTest.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        $app = new Application();
+        $app->add(new DobbleCommand());
+
+        $cmd = $app->find('dobble:run');
+        $cmd_tester = new CommandTester($cmd);
+        $cmd_tester->execute(array(
+            'command' => $cmd->getName(),
+            'elements_per_card' => 1,
+        ));
+    }
```
Comment (fzaninotto) [inline tests/DobbleCommandTest.php:16]: missing assertion here

File: composer.json [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
         "symfony/console": "^2.7"
+    },
+    "autoload-dev": {
+        "psr-4": {"": "tests/"}
+    },
+    "require-dev": {
+        "phpunit/phpunit": "^5.0"
+    },
+    "scripts": {
+        "test": ["phpunit tests/DobbleCommandTest"]
```
Comment (jpetitcolas) [inline composer.json]: Have a look on [phpunix.xml](https://github.com/symfony/symfony-standard/blob/2.8/app/phpunit.xml.dist) to test directly the whole `tests` folder.

File: tests/DobbleCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+use Dobble\DobbleCommand;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+
+class DobbleCommandTest extends \PHPUnit_Framework_TestCase
+{
+    public function testExecute()
+    {
+        $app = new Application();
```
Comment (fzaninotto) [inline tests/DobbleCommandTest.php]: move the first 4 lines to a `setUp()` method

Comment (jpetitcolas) [inline tests/DobbleCommandTest.php]: Move it into the `setUp` method.

File: tests/DobbleCommandTest.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        $app->add(new DobbleCommand());
+
+        $cmd = $app->find('dobble:run');
+        $cmd_tester = new CommandTester($cmd);
+        $cmd_tester->execute(array(
```
Comment (jpetitcolas) [inline tests/DobbleCommandTest.php]: Use short array notation:

``` php
$cmd_tester->execute([
    'command' => $cmd->getName(),
]);
```

File: tests/DobbleCommandTest.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        $cmd = $app->find('dobble:run');
+        $cmd_tester = new CommandTester($cmd);
+
+        // String are not accepted
+        $cmd_tester->execute(array(
+            'command' => $cmd->getName(),
+            'elements_per_card' => 'NaN',
+        ));
+
+        $this->assertRegExp(
```
Comment (jpetitcolas) [inline tests/DobbleCommandTest.php]: Write it on a single line

File: tests/DobbleCommandTest.php [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            '/must be a postive integer/',
+            $cmd_tester->getDisplay()
+        );
+
+        // Null value
+        $cmd_tester->execute(array(
```
Comment (jpetitcolas) [inline tests/DobbleCommandTest.php]: Use camelCase rule.

File: tests/DobbleCommandTest.php [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+            'elements_per_card' => 0,
+        ));
+
+        $this->assertRegExp(
+            '/must be a postive integer/',
+            $cmd_tester->getDisplay()
+        );
+
+        // Negative value
+        $cmd_tester->execute(array(
```
Comment (jpetitcolas) [inline tests/DobbleCommandTest.php]: Use dataProviders.

**Comment by fzaninotto** [comment]
Don't forget to add the Travis "build" image to the README to publicly show that the build is green!

**Pull request merged**